### ISSUE TITLE
Investigate the way to add an ability to change the severity level of custom lints in YAML file

### ISSUE THREAD

**Issue by solid-vovabeloded** [solid-software/solid_lints#79]
Currently, we can change the severity level using the [standard approach](https://dart.dev/tools/analysis#changing-the-severity-of-rules), but it produces a warning like on the screenshot below: 

<img width="591" alt="" src="https://github.com/solid-software/solid_lints/assets/41615621/75ee3e25-f421-4c7f-a161-6b1e1a949197">

We need to find a way to get rid of this warning or develop our own way of changing the severity level of custom lints.

**Comment by solid-illiaaihistov** [comment]
After migrating to analysis_server_plugin, the severity can be specified as follows:
```yaml
plugins:
  solid_lints:
    diagnostics:
      avoid_using_api: warning
```
More details here:
[[analyzer plugins] Support custom severity in analysis options](https://github.com/dart-lang/sdk/issues/59644)
[New YAML configuration for diagnostics analysis options](https://github.com/dart-lang/sdk/issues/57034)

**Comment by illia-romanenko** [comment]
@solid-illiaaihistov since it's done - can we close this?

**Comment by solid-illiaaihistov** [comment]
Yes, it's available in the 1.0.0-dev.1 release. I'm closing the task.
